frontend/apps/app/components/SessionDetailPage/components/Chat/components/TimelineItem/components/VersionMessage/VersionMessage.tsx

```diff
@@ -108,36 +131,69 @@ export const VersionMessage: FC<Props> = ({ buildingSchemaVersionId }) => {
   }
 
   return (
-    <div className={styles.container}>
-      <button type="button" className={styles.header} onClick={toggleExpanded}>
-        <div
-          className={clsx(styles.chevron, isExpanded ? styles.expanded : '')}
-        >
-          <ChevronDown />
+    <div className={clsx(styles.container, isExpanded && styles.expanded)}>
+      <button
+        type="button"
+        className={clsx(styles.header, isExpanded && styles.expanded)}
+        onClick={toggleExpanded}
+      >
+        <div className={styles.collapseButton}>
+          {isExpanded ? <ChevronDown /> : <ChevronRight />}
         </div>
         <span className={styles.versionNumber}>
           Version {displayVersionNumber}
         </span>
       </button>
 
-      <div className={clsx(styles.content, isExpanded ? styles.expanded : '')}>
-        <div className={styles.operationList}>
+      <div className={styles.divider} />
+      <div
+        className={clsx(styles.contentWrapper, isExpanded && styles.expanded)}
+      >
+        <div className={styles.content}>
           {patchOperations.map((operation, index) => (
             <div
               key={`${version.id}-${index}`}
-              className={styles.operationItem}
+              className={clsx(
+                styles.operationItem,
+                styles[operation.statusClass],
+              )}
             >
-              <span className={styles.operationName}>{operation.path}</span>
-              <span className={styles.operationStatus}>{operation.status}</span>
+              <div className={styles.pathContainer}>
+                {operation.path.map((part, partIndex) => (
+                  // biome-ignore lint/suspicious/noArrayIndexKey: Path parts maintain their order
+                  <Fragment key={`${part}-${partIndex}`}>
```

Comment:
[IMO]
I think it will be unique if you use **buildingSchemaVersionId**.

```suggestion
                  <Fragment key={`${buildingSchemaVersionId}-${partIndex}`}>
```